docs/designs/xref.md

```diff
@@ -0,0 +1,101 @@
+# Xref
+- User can define uid with some properties in markdown, JSON and YAML files. Then output xrefmap.json for others to reference to. 
```

Comment:
better give some explanation why we use `uid`, not relative path/absolute path/url?